Automattic/jetpack
# Search: Initialize debug bar using correct class

#### Changes proposed in this Pull Request:
Refactors `Jetpack_Search_Debug_Bar`'s render function to conditionally use either the `Classic_Search` singleton or the `Instant_Search` singleton.

#### Jetpack product discussion
p8oabR-Nm-p2#comment-5939

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your site with the [Debug Bar](https://wordpress.org/plugins/debug-bar/) and [Query Monitor](https://wordpress.org/plugins/query-monitor/) plugins.
* Purchase a Jetpack Search subscription.
* Ensure that the Jetpack wp-admin dashboard works as expected.
* Ensure that the Search dashboard works as expected (`/wp-admin/admin.php?page=jetpack-search`).
